AppTaskQueue/appscale/taskqueue/queue.py

```diff
@@ -248,24 +248,26 @@ class PostgresPullQueue(Queue):
 
   TTL_INTERVAL_AFTER_DELETED = '7 days'
 
-  def __init__(self, queue_info, app, pg_connection=None):
+  def __init__(self, queue_info, app, pg_connection_pool):
     """ Create a PostgresPullQueue object.
 
     Args:
       queue_info: A dictionary containing queue info.
       app: A string containing the application ID.
-      pg_connection: A psycopg2 connection to PostgreSQL.
+      pg_connection_pool: A psycopg2 connection pool.
     """
     from psycopg2 import IntegrityError  # Import psycopg2 lazily
     super(PostgresPullQueue, self).__init__(queue_info, app)
-    self.pg_connection = pg_connection
+    self.connection_key = self.app
```

Comment:
@cdonati , this PR is aimed to make Taskqueue resistant to postgres connection problems.
Initially I tried to implement connection wrapper which would do reconnection automatically, but that solution wouldn't work with concurrent requests.
Then I decided that client (PostgresPullQueue) shouldn't hold connection or connection wrapped on its own, but it should always ask workable connection from external object (ConnectionPool).

This modified connection pool takes care about reconnect.

ConnectionPool is created for every project. All queues in the project uses single connection (`self.connection_key` is the way how PostgresPullQueue refers to this connection). When the project is deleted, all connections (single connection in this case) are closed.

Reply:
That makes sense. Thanks for the explanation.